File: library/src/main/java/com/sloydev/preferator/editor/FloatPrefEditor.java
package com.sloydev.preferator.editor;


import android.content.Context;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.EditText;
import android.widget.FrameLayout;

import com.sloydev.preferator.R;

public class FloatPrefEditor extends FrameLayout {

    private EditText valueView;
    private OnFloatValueChangeListener onFloatValueChangeListener;

    public FloatPrefEditor(Context context) {
        this(context, null);
    }

    public FloatPrefEditor(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public FloatPrefEditor(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        LayoutInflater.from(getContext()).inflate(R.layout.item_editor_float, this, true);
        valueView = (EditText) findViewById(R.id.pref_value);
        valueView.addTextChangedListener(new TextWatcher() {
            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                if (onFloatValueChangeListener != null) {
                    try {
                        Float number;
                        if (charSequence != null) {
                            number = Float.parseFloat(charSequence.toString());
                        } else {
                            number = 0f;
                        }
                        onFloatValueChangeListener.onValueChange(number);
                        valueView.setError(null);
                    } catch (NumberFormatException e) {
                        valueView.setError("Wrong integer format");
                    }
                }
            }

            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override
            public void afterTextChanged(Editable editable) {

            }
        });
    }

    public void setValue(Float value) {
        valueView.setText(value.toString());
    }

    public Float getValue() {
        return Float.parseFloat(valueView.getText().toString());
    }

    public void setOnFloatValueChangeListener(OnFloatValueChangeListener onLongValueChangeListener) {
        this.onFloatValueChangeListener = onLongValueChangeListener;
    }

    public interface OnFloatValueChangeListener {
        void onValueChange(Float newValue);
    }
}

File: library/src/main/java/com/sloydev/preferator/editor/FloatPrefEditor.kt
package com.sloydev.preferator.editor

import android.content.Context
import android.text.Editable
import android.text.TextWatcher
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.EditText
import android.widget.FrameLayout
import com.sloydev.preferator.R

class FloatPrefEditor @JvmOverloads constructor(
  context: Context?,
  attrs: AttributeSet? = null,
  defStyleAttr: Int = 0
) : FrameLayout(context, attrs, defStyleAttr) {
  private var valueView: EditText? = null
  private var onFloatValueChangeListener: OnFloatValueChangeListener? = null
  private fun init() {
    LayoutInflater.from(context).inflate(R.layout.item_editor_float, this, true)
    valueView = findViewById(R.id.pref_value) as EditText
    valueView!!.addTextChangedListener(object : TextWatcher {
      override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
        if (onFloatValueChangeListener != null) {
          try {
            val number: Float
            number = charSequence?.toString()?.toFloat() ?: 0f
            onFloatValueChangeListener!!.onValueChange(number)
            valueView!!.error = null
          } catch (e: NumberFormatException) {
            valueView!!.error = "Wrong integer format"
          }
        }
      }

      override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
      override fun afterTextChanged(editable: Editable) {}
    })
  }

  var value: Float
    get() = valueView!!.text.toString().toFloat()
    set(value) {
      valueView!!.setText(value.toString())
    }

  fun setOnFloatValueChangeListener(onLongValueChangeListener: OnFloatValueChangeListener?) {
    onFloatValueChangeListener = onLongValueChangeListener
  }

  interface OnFloatValueChangeListener {
    fun onValueChange(newValue: Float?)
  }

  init {
    init()
  }
}